File: Models/SensorApplication.cs

using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;
using Microsoft.AspNetCore.Mvc.ModelBinding;

namespace RoomMonitor.Models
{
    public class SensorApplication
    {
        [Required]
        public string Application { get; set; }

        [Required]
        public string Name { get; set; }

        [Required]
        public string Description { get; set; }
    }

    public class ApplicationSwitch : SensorApplication
    {

        [Required]
        public string Value_0 { get; set; }

        [Required]
        public string Value_1 { get; set; }

    }

    public class ApplicationMeasure : SensorApplication
    {

        [Required]
        public string UnitMeasure { get; set; }

        public decimal? LimitMin { get; set; }
        public decimal? LimitMax { get; set; }
        public decimal? DefaultMin { get; set; }
        public decimal? DefaultMax { get; set; }

    }
}

File: ClientApp/src/app/shared/models/SensorApplication.ts
export interface ISwitchApplication {
  name: string;
  description: string;
  value_0: string;
  value_1: string;
}

export interface IMeasureApplication {
  name: string;
  description: string;
  unitMeasure: string;
  limitMin?: number;
  limitMax?: number;
  defaultMin?: number;
  defaultMax?: number;
}
